flink-runtime/src/main/java/org/apache/flink/runtime/operators/sort/NormalizedKeySorter.java

```diff
@@ -296,7 +298,13 @@ public boolean write(T record) throws IOException {
         try {
             this.serializer.serialize(record, this.recordCollector);
         } catch (EOFException e) {
+            LOG.error("Out of memory");
             return false;
+        } catch (Exception e) {
+            if (ExceptionUtils.indexOfType(e.getCause(), EOFException.class) != -1) {
+                LOG.error("Out of memory");
+                return false;
+            }
```

Comment:
shouldn't we throw the exception in the else block (looks like its absorbed otherwise)?

Reply:
yeah, actually +1 question on that